### PR TITLE
Support VAllocation: initial commit

### LINKED ISSUE #965: Forma: Asignación de Pagos (Payment Allocation)
Image

- [x] solop-develop/vue-1.0#2074
- [x] Forma: Asignación de Pagos > Listar Organizaciones de filtro.
- [x] solop-develop/vue-1.0#2036
- [x] solop-develop/frontend-core#128
- [x] #2071
- [x] Forma: Asignación de Pagos > Listar Facturas.
- [x] Forma: Asignación de Pagos > Listar Cargos.
- [x] Forma: Asignación de Pagos > Listar Organizaciones de sobre-escritura de transacción.
- [x] Forma: Asignación de Pagos > Calcular Diferencia.
- [x] Forma: Asignación de Pagos > Procesar.

### BugFIx 

- [x] solop-develop/vue-1.0#1312
- [x] solop-develop/vue-1.0#1313
- [x] solop-develop/vue-1.0#1314
- [x] solop-develop/vue-1.0#1315
- [x] solop-develop/vue-1.0#1371
- [ ] solop-develop/frontend-core#127
- [ ] solop-develop/frontend-core#129
- [x] solop-develop/vue-1.0#2076

### PR DESCRIPTION
**Pull request by elsiosanchez**
https://user-images.githubusercontent.com/45974454/210832520-cd89098e-9639-42e7-98ce-82742647ec94.mp4

fixes https://github.com/solop-develop/frontend-core/issues/965

### CODE REVIEW

File: components/ADempiere/Form/VAllocation/components/Invoce.vue [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+                  label="Cantidad de Registro:"
+                >
+                  <b>
+                    5
+                  </b>
+                </el-form-item>
+              </el-col>
+              <el-col :span="24">
+                <el-form-item
+                  label="Total Factura:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Invoce.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Invoce.vue [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                >
+                  <b>
+                    8.506,23
+                  </b>
+                </el-form-item>
+              </el-col>
+              <el-col :span="24">
+                <el-form-item
+                  label="Convertido:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Invoce.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Invoce.vue [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        >
+          <div>
+            <el-form
+              :inline="true"
+              label-position="left"
+              style="padding: 10px !important;"
+            >
+              <el-col :span="24">
+                <el-form-item
+                  label="Cantidad de Registro:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Invoce.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Invoce.vue [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+        :key="key"
+        prop="id"
+        :type="header.type"
+        :label="header.label"
+      />
+    </el-table>
+    <el-card class="box-card">
+      <div slot="header" class="clearfix" style="text-align: center;">
+        <b>
+          {{ 'Detalles de la Factura' }}
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Invoce.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Invoce.vue [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+        >
+          <div>
+            <el-form
+              :inline="true"
+              label-position="left"
+              style="padding: 10px !important;"
+            >
+              <el-col :span="24">
+                <el-form-item
+                  label="Pagado:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Invoce.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Invoce.vue [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+                >
+                  <b>
+                    350,63
+                  </b>
+                </el-form-item>
+              </el-col>
+              <el-col :span="24">
+                <el-form-item
+                  label="Asignado:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Invoce.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Invoce.vue [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+                >
+                  <b>
+                    170,00
+                  </b>
+                </el-form-item>
+              </el-col>
+              <el-col :span="24">
+                <el-form-item
+                  label="Monto Abierto:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Invoce.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Invoce.vue [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+    </el-card>
+  </div>
+</template>
+
+<script>
+import { defineComponent, ref } from '@vue/composition-api'
+// import lang from '@/lang'
+// components and mixins
+import Carousel from '@theme/components/ADempiere/Carousel'
+// import formMixin from '@theme/components/ADempiere/Form/formMixin.js'
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Invoce.vue]: remove unused imports

File: components/ADempiere/Form/VAllocation/components/Payments.vue [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+        >
+          <div>
+            <el-form
+              :inline="true"
+              label-position="left"
+              style="padding: 10px !important;"
+            >
+              <el-col :span="24">
+                <el-form-item
+                  label="Cantidad de Registro:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Payments.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Payments.vue [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+                >
+                  <b>
+                    5
+                  </b>
+                </el-form-item>
+              </el-col>
+              <el-col :span="24">
+                <el-form-item
+                  label="Total Factura:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Payments.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Payments.vue [hunk 11]
```diff
@@ -0,0 +1,277 @@
...
+                >
+                  <b>
+                    8.506,23
+                  </b>
+                </el-form-item>
+              </el-col>
+              <el-col :span="24">
+                <el-form-item
+                  label="Convertido:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Payments.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Payments.vue [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+        >
+          <div>
+            <el-form
+              :inline="true"
+              label-position="left"
+              style="padding: 10px !important;"
+            >
+              <el-col :span="24">
+                <el-form-item
+                  label="Pagado:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Payments.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Payments.vue [hunk 13]
```diff
@@ -0,0 +1,277 @@
...
+                >
+                  <b>
+                    350,63
+                  </b>
+                </el-form-item>
+              </el-col>
+              <el-col :span="24">
+                <el-form-item
+                  label="Asignado:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Payments.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Payments.vue [hunk 14]
```diff
@@ -0,0 +1,277 @@
...
+                >
+                  <b>
+                    170,00
+                  </b>
+                </el-form-item>
+              </el-col>
+              <el-col :span="24">
+                <el-form-item
+                  label="Monto Abierto:"
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Payments.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/Payments.vue [hunk 15]
```diff
@@ -0,0 +1,277 @@
...
+    </el-card>
+  </div>
+</template>
+
+<script>
+import { defineComponent, ref } from '@vue/composition-api'
+// import lang from '@/lang'
+// components and mixins
+import Carousel from '@theme/components/ADempiere/Carousel'
+// import formMixin from '@theme/components/ADempiere/Form/formMixin.js'
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Payments.vue]: remove unused comments

File: components/ADempiere/Form/VAllocation/components/Payments.vue [hunk 16]
```diff
@@ -0,0 +1,277 @@
...
+        columnName: '',
+        width: '150'
+      },
+      {
+        label: 'Monto Asignado',
+        columnName: '',
+        width: '150'
+      },
+      {
+        label: 'Monto Abierto',
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Payments.vue]: @elsiosanchez please add translation.

**Review by EdwinBetanc0urt — CHANGES_REQUESTED** [review]

File: components/ADempiere/Form/VAllocation/components/Payments.vue (hunk 16 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Payments.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Invoce.vue (hunk 1 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Invoce.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Invoce.vue (hunk 2 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Invoce.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Invoce.vue (hunk 3 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Invoce.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Invoce.vue (hunk 5 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Invoce.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Payments.vue (hunk 12 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Payments.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Payments.vue (hunk 13 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Payments.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Payments.vue (hunk 14 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Payments.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Payments.vue (hunk 15 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Payments.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Payments.vue (hunk 16 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Payments.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Invoce.vue (hunk 4 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Invoce.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Invoce.vue (hunk 6 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Invoce.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Invoce.vue (hunk 7 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Invoce.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Payments.vue (hunk 11 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Payments.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Payments.vue (hunk 10 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Payments.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Payments.vue (hunk 9 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Payments.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Invoce.vue (hunk 8 above)
Comment (elsiosanchez) [reply components/ADempiere/Form/VAllocation/components/Invoce.vue]: Added the translation in the following commit [68211fe](https://github.com/solop-develop/frontend-default-theme/pull/442/commits/68211fe355873ffe49725eb888013312491e1af4)

File: components/ADempiere/Form/VAllocation/components/Payments.vue [hunk 17]
```diff
@@ -0,0 +1,331 @@
...
+along with this program. If not, see <https:www.gnu.org/licenses/>.
+-->
+
+<template>
+  <div style="display: contents;height: 100% !important;">
+    <el-card id="panel-top-search-criteria" class="panel-top-search-criteria">
+      <div style="width: 50%;">
+        <el-card style="padding: 5px 10px 5px 10px;">
+          <div slot="header" class="clearfix" style="text-align: center;">
+            <b> {{ 'Pagos' }} </b>
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Payments.vue]: @elsiosanchez please add translation

File: components/ADempiere/Form/VAllocation/components/Payments.vue [hunk 18]
```diff
@@ -0,0 +1,331 @@
...
+              :width="header.width"
+              :label="header.label"
+            />
+          </el-table>
+        </el-card>
+      </div>
+      <div style="width: 50%;">
+        <el-card style="padding: 5px 10px 5px 10px;">
+          <div slot="header" class="clearfix" style="text-align: center;">
+            <b> {{ 'Facturas' }} </b>
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/Payments.vue]: @elsiosanchez please add translation.

File: components/ADempiere/Form/VAllocation/components/SearchCriteria.vue [hunk 19]
```diff
@@ -0,0 +1,488 @@
...
+  </div>
+</template>
+
+<script>
+import { defineComponent, ref, computed } from '@vue/composition-api'
+import store from '@/store'
+// import lang from '@/lang'
+// components and mixins
+import Carousel from '@theme/components/ADempiere/Carousel'
+// import formMixin from '@theme/components/ADempiere/Form/formMixin.js'
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/SearchCriteria.vue:181]: @elsiosanchez remove unused comments

File: components/ADempiere/Form/VAllocation/components/SearchCriteria.vue [hunk 20]
```diff
@@ -0,0 +1,488 @@
...
+
+    const currentTypeTransaction = computed({
+      // getter
+      get() {
+        const { transactionType } = store.getters.getSearchFilter
+        return transactionType
+      },
+      // setter
+      set(type) {
+        console.log({ type })
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/SearchCriteria.vue:306]: remove console.log

File: components/ADempiere/Form/VAllocation/components/SearchCriteria.vue [hunk 21]
```diff
@@ -0,0 +1,488 @@
...
+
+    const currentDate = computed({
+      // getter
+      get() {
+        const { date } = store.getters.getSearchFilter
+        return date
+      },
+      // setter
+      set(date) {
+        console.log({ date })
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/components/SearchCriteria.vue:293]: remove console.log

File: components/ADempiere/Form/VAllocation/index.vue [hunk 22]
```diff
@@ -0,0 +1,143 @@
...
+      {
+        name: lang.t('form.VAllocation.step.summaryAndAdjustment'),
+        key: 'summary'
+      }
+    ])
+
+    const currentSetp = ref(0)
+
+    function nextStep(step) {
+      console.log(currentSetp.value)
```
Comment (EdwinBetanc0urt) [inline components/ADempiere/Form/VAllocation/index.vue:109]: remove console.log

**Review by EdwinBetanc0urt — CHANGES_REQUESTED** [review]

**Pull request merged**